### ISSUE TITLE
detect POWER9

### ISSUE THREAD

**Issue by jeffhammond** [flame/blis#322]
BLIS should attempt to autodetect POWER9 via `lscpu`.  I'm sure there is a nice way to do this with Linux shell commands but I haven't tried to write it.

```
[jrhammon@power9-vm6 blis]$ ./configure
configure: detected Linux kernel version 4.14.0-115.8.1.el7a.ppc64le.
configure: python interpeter search list is: python python3 python2.
configure: using 'python' python interpreter.
configure: found python version 2.7.5 (maj: 2, min: 7, rev: 5).
configure: python 2.7.5 appears to be supported.
configure: C compiler search list is: gcc clang cc.
configure: using 'gcc' C compiler.
configure: C++ compiler search list is: g++ clang++ c++.
configure: using 'g++' C++ compiler (for sandbox only).
configure: found gcc version 4.8.5 (maj: 4, min: 8, rev: 5).
configure: checking for blacklisted configurations due to gcc 4.8.5.
configure: warning: gcc 4.8.5 does not support 'knl'; adding to blacklist.
configure: warning: gcc 4.8.5 does not support 'excavator'; adding to blacklist.
configure: warning: gcc 4.8.5 does not support 'zen'; adding to blacklist.
configure: warning: gcc 4.8.5 does not support 'skx'; adding to blacklist.
configure: found assembler ('as') version 2.27 (maj: 2, min: 27, rev: ).
configure: checking for blacklisted configurations due to as 2.27.
configure: warning: assembler ('as' 2.27) does not support 'bulldozer'; adding to blacklist.
configure: warning: assembler ('as' 2.27) does not support 'sandybridge'; adding to blacklist.
configure: warning: assembler ('as' 2.27) does not support 'haswell'; adding to blacklist.
configure: warning: assembler ('as' 2.27) does not support 'piledriver'; adding to blacklist.
configure: warning: assembler ('as' 2.27) does not support 'steamroller'; adding to blacklist.
configure: configuration blacklist:
configure:   knl excavator zen skx bulldozer sandybridge haswell piledriver steamroller
configure: reading configuration registry...done.
configure: determining default version string.
configure: found '.git' directory; assuming git clone.
configure: executing: git describe --tags.
configure: git returned an error: './configure: line 1627: git: command not found'.
configure: using string from unmodified version file.
configure: starting configuration of BLIS 0.6.0.
configure: configuring with official version string.
configure: found shared library .so version '2.0.0'.
configure:   .so major version: 2
configure:   .so minor.build version: 0.0
configure: 
configure: *** No configuration given! ***
configure: 
configure: Default configuration behavior is not implemented (for your
configure: own safety). Please re-run 'configure' and specify one
configure: of the existing configurations in the source distribution's
configure  'config_registry' file:
configure: 
configure:   amd64 (generic)
configure:   bgq (bgq)
configure:   cortexa15 (cortexa15)
configure:   cortexa53 (cortexa53)
configure:   cortexa57 (cortexa57)
configure:   cortexa9 (cortexa9)
configure:   generic (generic)
configure:   intel64 (penryn generic)
configure:   penryn (penryn)
configure:   power9 (power9)
configure:   thunderx2 (thunderx2)
configure:   x86_64 (penryn generic)
configure: 
```

```
[jrhammon@power9-vm6 blis]$ lscpu
Architecture:          ppc64le
Byte Order:            Little Endian
CPU(s):                40
On-line CPU(s) list:   0-39
Thread(s) per core:    1
Core(s) per socket:    1
Socket(s):             40
NUMA node(s):          1
Model:                 2.1 (pvr 004e 1201)
Model name:            POWER9 (architected), altivec supported
Hypervisor vendor:     KVM
Virtualization type:   para
L1d cache:             32K
L1i cache:             32K
NUMA node0 CPU(s):     0-39
```

**Comment by loveshack** [comment]
At least with GCC, use __builtin_cpu_is to distinguish POWER micro-architectures.

**Comment by fgvanzee** [comment]
Note that @nicholaiTukanov is working on power9, but I'm guessing this is probably quite low on the list of things he would opt to work on right now. If this is going to be done anytime soon, someone else will probably need to step up and suggest a solution. In other words... "We look forward to your patch." :)

**Comment by loveshack** [comment]
> If this is going to be done anytime soon, someone else will
> probably need to step up and suggest a solution. In other words... "We
> look forward to your patch." :)

Again, I don't understand as I've given a solution similarly to aarch64,
and it was Jeff who proposed it originally.  I don't have POWER to hand
either.

**Comment by fgvanzee** [comment]
@loveshack My message was directed at Jeff, especially because he taught me the quoted phrase, but also because he was the original poster. Sorry for any confusion I may have caused.

As for your solution, I wouldn't quite call it a solution for our purposes. 1. We like PRs. 2. We like portability, and I would hesitate to use a gcc builtin unless that was the only way. But like I said before, we don't have the resources to work on this right now. We will keep you (both) updated if/when we get the opportunity to look at this. As always, thank you for your contributions.

**Comment by devinamatthews** [comment]
I think this can be closed given #345 unless @jeffhammond wants to submit a separate PR.
